doc/cygnus-ngsi/integration/orion_cygnus_spark.md

```diff
@@ -0,0 +1,216 @@
+
+```
+$ cat /usr/cygnus/conf/agent_spark.conf
+cygnusagent.sources = http-source
+cygnusagent.sinks = avro-sink
+cygnusagent.channels = avro-channel
+
+cygnusagent.sources.http-source.type = http
+cygnusagent.sources.http-source.channels = avro-channel
+cygnusagent.sources.http-source.port = 5050
+cygnusagent.sources.http-source.handler = com.telefonica.iot.cygnus.handlers.NGSIRestHandler
+cygnusagent.sources.http-source.handler.notification_target = /notify
+cygnusagent.sources.http-source.handler.default_service = default
+cygnusagent.sources.http-source.handler.default_service_path = /
+
+cygnusagent.sinks.avro-sink.type = avro
+cygnusagent.sinks.avro-sink.channel = avro-channel
+cygnusagent.sinks.avro-sink.hostname = 0.0.0.0
+cygnusagent.sinks.avro-sink.port = 5051
+
+cygnusagent.channels.avro-channel.type = memory
+cygnusagent.channels.avro-channel.capacity = 1000
+cygnusagent.channels.avro-channel.transactionCapacity = 100
+```
+
+A Cygnus instance configuration file must be added too, somethinkg like:
+
+```
+$ cat /usr/cygnus/conf/cygnus_instance_spark.conf
+CYGNUS_USER=cygnus
+CONFIG_FOLDER=/usr/cygnus/conf
+CONFIG_FILE=/usr/cygnus/conf/agent_spark.conf
+AGENT_NAME=cygnus-ngsi
+LOGFILE_NAME=cygnus.log
+ADMIN_PORT=8081
+POLLING_INTERVAL=30
+```
+
+Then, it can be starte as a service:
+
+```
+$ sudo service cygnus start
+Starting Cygnus spark...                                    [  OK  ]
+```
+
+In order to test the integracion among Orion Context Broker and Cygnus, simply update any context entity and check the related notification reception in Cygnus logs:
+
+```
+$ tail -f /var/log/cygnus/cygnus.log
+...
+time=2017-01-27T13:33:28.236UTC | lvl=INFO | corr=a72db6ff-ce15-4d38-a3a3-95af2ecc73b8 | trans=a72db6ff-ce15-4d38-a3a3-95af2ecc73b8 | srv=sc_valencia | subsrv=/gardens | comp=cygnus-ngsi | op=getEvents | msg=com.telefonica.iot.cygnus.handlers.NGSIRestHandler[304] : [NGSIRestHandler] Received data ({  "subscriptionId" : "51c0ac9ed714fb3b37d7d5a8",  "originator" : "localhost",  "contextResponses" : [    {      "contextElement" : {        "attributes" : [          {            "name" : "temperature",            "type" : "centigrade",            "value" : "26.5"          }        ],        "type" : "room",        "isPattern" : "false",        "id" : "room1"      },      "statusCode" : {        "code" : "200",        "reasonPhrase" : "OK"      }    }  ]})
+...
+```
+
+[Top](#top)
+
+##<a name="section4"></a>Setting up spark Streaming
+Apache Spark will be installed from the URL resulting from entering at [Spark Download page](http://spark.apache.org/downloads.html) and selecting:
+
+* Spark release: 1.6.3
+* Package type: pre-built for Hadoop 2.3
+* Download type: direct download
+
+The URL the "download Spark" link points to must be copied and then downloaded in our machine (into our home dir, for instance) by doing:
```

Comment:
I'd suggest to add a "(replace d3kbcqa49mib13.cloudfront.net by the actual hostname in your case)" to be prepared in the case of future change.